packages/xchain-radix/src/client.ts

```diff
@@ -0,0 +1,465 @@
+          requestBody = this.constructRequestBody(params, response.data.next_cursor)
+          response = await axios.post(url, requestBody)
+        } else {
+          break
+        }
+      }
+
+      return { total: transactions.length, txs: transactions }
+    } catch (error) {
+      throw new Error('Failed to get transactions')
+    }
+  }
+
+  private constructRequestBody(params: TxHistoryParams, cursor?: string) {
+    const requestBody: any = {
+      affected_global_entities_filter: [params.address],
+      limit_per_page: params.limit && params.limit > 100 ? 100 : params.limit,
+      manifest_resources_filter: [params.asset],
+      from_ledger_state: {
+        state_version: params.offset,
+      },
+      opt_ins: {
+        raw_hex: true,
+      },
+    }
+
+    if (cursor) {
+      requestBody.cursor = cursor
+    }
+
+    return requestBody
+  }
+
+  /**
+   * Get the transaction details of a given transaction id.
+   * This method uses LTSRadixEngineToolkit.Transaction.summarizeTransaction
+   * to convert a transaction hex to a transaction summary. If the transaction was not built with
+   * the SimpleTransactionBuilder, the method will fail to get the transaction data
+   * @param {string} txId The transaction id.
+   * @returns {Tx} The transaction details of the given transaction id.
+   */
+  async getTransactionData(txId: string): Promise<Tx> {
+    const url = `${this.getGatewayUrl()}${TRANSACTION_COMMITTED_DETAILS_PATH}`
+    const requestBody = {
+      intent_hash: txId,
+      opt_ins: {
+        manifest_instructions: true,
+        raw_hex: true,
+      },
+    }
+
+    try {
+      const response = await axios.post(url, requestBody)
+      const transaction: Tx = await this.convertTransactionFromHex(
+        response.data.transaction.raw_hex,
+        response.data.transaction.confirmed_at,
+        response.data.transaction.intent_hash,
+      )
+      return transaction
+    } catch (error) {
+      throw new Error('Failed to fetch transaction data')
+    }
+  }
+
+  /**
+   * Helper function to convert a transaction in hex, returned by the gateway to a Tx type
+   * @param transaction_hex - The raw_hex returned by the gateway for a transaction id
+   * @param confirmed_at - The confirmed_at date for the transaction
+   * @param intent_hash - The transaction intent hash
+   * @returns a transaction in Tx type
+   */
+  async convertTransactionFromHex(transaction_hex: string, confirmed_at: string, intent_hash: string): Promise<Tx> {
+    const binaryString = Buffer.from(transaction_hex, 'hex').toString('binary')
+    const transactionBinary = new Uint8Array(binaryString.split('').map((char) => char.charCodeAt(0)))
+    const transactionSummary = await LTSRadixEngineToolkit.Transaction.summarizeTransaction(transactionBinary)
+    const withdrawAccount = Object.keys(transactionSummary.withdraws)[0]
+    const depositAccount = Object.keys(transactionSummary.deposits)[0]
```

Comment:
This is wrong. We need better error handling to be sure we never return incorrect data.

e.g. I don't think anything guarantees the `withdraws` and `deposits` are the same resource address?

Reply:
In general, we should only return an `asset` if the transaction is guaranteed to be a transfer of a single asset between two accounts.

I imagine we may want to handle these cases:
* The transactionSummary identifies it as a transfer between accounts  or `general` AND it's a transfer of one asset between accounts/components - use below
* The transactionSummary identifies it as a transfer between accounts or or `general` AND it's a transfer of more than one asset between accounts/components - possibly fill in `from`/`to` below, but don't fill in the `asset`. To be spec compliant, we may need some "fallback" asset, e.g. `N/A - More than one asset` or something. We probably need to check with someone what to do here.
* Otherwise, we need some fallback for the whole transaction. Maybe we should throw? Maybe we should find some base error model that fits with the schema? We should ideally ask someone how to handle this.

Reply:
Whilst it's important that it works in this use case, we should code defensively to ensure that when used on any transaction, it doesn't give incorrect data. In these failure cases we should do something to make it clear we can't parse it. Perhaps it's throw an error, but likely better is to return some error in the model. Otherwise `getTransactions` might just throw because of 1 non-conforming transaction in the user's history.

Reply:
@dhedey my intention with the current implementation is to log those transactions that could not be parsed and return the ones that can be parsed.